R/mobr.R

```diff
@@ -240,9 +240,8 @@ deltaS_N = function(comm, ref_dens, inds){
     # rescale and interpolate this plot based S to individual based 
     # using the ref_density (i.e., not the observed density)
     rescaled_effort = round(1:nplots * ref_dens)
-    if (max(rescaled_effort) < max(inds))
-        warning('Extrapolating the rarefaction curve because the number of rescaled individuals is smaller than the inds argument')
-    interp_S_samp = pchip(c(1, rescaled_effort), c(1, S_samp), inds)
+    # No extrapolation of the rescaled rarefaction curve, only interpolation
+    interp_S_samp = pchip(c(1, rescaled_effort), c(1, S_samp), inds[inds <= max(rescaled_effort)])[1:length(inds)]
```

Comment:
Nice job fixing this! I can see why this code works but it will difficult to understand for future us and other users. If we don't know that R adds NA's to cells that are longer than the vector is defined across we would be scratching our heads right now. I would prefer one step where you define `interp_S_samp` and then a second step where you add NA's to that vector so that it has the same length as `inds`.  What do you think?